x/ibc/04-channel/keeper/keeper.go

```diff
@@ -62,20 +65,24 @@ func (k Keeper) SetChannel(ctx sdk.Context, portID, channelID string, channel ty
 }
 
 // GetChannelCapability gets a channel's capability key from the store
-func (k Keeper) GetChannelCapability(ctx sdk.Context, portID, channelID string) (string, bool) {
+func (k Keeper) GetChannelCapability(ctx sdk.Context, portID, channelID string) (capability.Capability, bool) {
 	store := ctx.KVStore(k.storeKey)
 	bz := store.Get(ibctypes.KeyChannelCapabilityPath(portID, channelID))
 	if bz == nil {
-		return "", false
+		return nil, false
 	}
 
-	return string(bz), true
+	var capKey capability.Capability
+	k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, &capKey)
+
+	return capKey, true
 }
 
 // SetChannelCapability sets a channel's capability key to the store
-func (k Keeper) SetChannelCapability(ctx sdk.Context, portID, channelID string, key string) {
+func (k Keeper) SetChannelCapability(ctx sdk.Context, portID, channelID string, key capability.Capability) {
```

Comment:
Here again, we cannot be storing dynamic capabilities in state. What is the true intended behavior here?

Should these be wrappers over setting on the scopedKeeper? Or should they be storing static capabilities?

Reply:
This function can be completely removed